src/app/user/components/favorite-products/favorite-products.component.html

```diff
@@ -0,0 +1,14 @@
+<div *ngIf="(products$ | async).length; else elseBlock">
```

Comment:
Can we use a default value here as an empty array. So that we don't do length check here.
Just `*ngIf="products$ | async; let products; else elseBlock"` would be enough.

Reply:
@pkrawat1  if we use this, then empty array would always return true and else block will never run.

Reply:
```html
<div *ngIf="products$ | async; let products">
  <table class="table" *ngIf="products.length; else elseBlock">
    <thead class="thead-light">
      <tr>
        <th scope="col">Product Image</th>
        <th scope="col">Product Name</th>
        <th scope="col">Action</th>
      </tr>
    </thead>
    <tbody>
      <app-favorite-product-list-item *ngFor="let product of products" [product]="product"></app-favorite-product-list-item>
    </tbody>
  </table>

  <ng-template #elseBlock>
    <div>You have'nt favorited anything yet.</div>
  </ng-template>
</div>
```